### PR TITLE
Configure templates with default logback.xml template

### PR DESCRIPTION
**Pull request by wsargent**
The documentation points to a logback.xml configuration that uses an ASYNCFILE / ASYNCSTDOUT appender, but the tempates use straight stdout and no file.  This is confusing -- they should all start from the same place, and there's no reason (that I can tell) that you'd want to remove file logging.

https://www.playframework.com/documentation/2.4.x/SettingsLogger#Default-configuration

### CODE REVIEW

**Comment by wsargent** [comment]
[error]     ! clean up when closed

https://travis-ci.org/playframework/playframework/jobs/96846856#L3997

**Comment by wsargent** [comment]
https://travis-ci.org/playframework/playframework/jobs/97152221#L2069

File: templates/play-java-intro/conf/logback.xml [hunk 1]
```diff
@@ -1,22 +1,41 @@
...
   <logger name="play" level="INFO" />
   <logger name="application" level="DEBUG" />
 
-  <root level="ERROR">
-    <appender-ref ref="STDOUT" />
+  <!-- Off these ones as they are annoying, and anyway we manage configuration ourself -->
+  <logger name="com.avaje.ebean.config.PropertyMapLoader" level="OFF" />
+  <logger name="com.avaje.ebeaninternal.server.core.XmlConfigLoader" level="OFF" />
+  <logger name="com.avaje.ebeaninternal.server.lib.BackgroundThread" level="OFF" />
+  <logger name="com.gargoylesoftware.htmlunit.javascript" level="OFF" />
```
Comment (benmccann) [inline templates/play-java-intro/conf/logback.xml]: we should really have an example of configuring logging separately for tests (i.e. probably put this line in a new file in `test/resources/logback-test.xml`)

File: templates/play-java-intro/conf/logback.xml [hunk 2]
```diff
@@ -1,22 +1,41 @@
...
+    <appender-ref ref="STDOUT" />
+  </appender>
+
   <logger name="play" level="INFO" />
   <logger name="application" level="DEBUG" />
 
-  <root level="ERROR">
-    <appender-ref ref="STDOUT" />
+  <!-- Off these ones as they are annoying, and anyway we manage configuration ourself -->
+  <logger name="com.avaje.ebean.config.PropertyMapLoader" level="OFF" />
```
Comment (benmccann) [inline templates/play-java-intro/conf/logback.xml]: it'd be nice to file a bug with ebean to let them know their default logging is annoying and reference it here

File: templates/play-scala/conf/logback.xml [hunk 3]
```diff
@@ -1,22 +1,41 @@
...
+    <appender-ref ref="STDOUT" />
+  </appender>
+
   <logger name="play" level="INFO" />
   <logger name="application" level="DEBUG" />
 
-  <root level="ERROR">
-    <appender-ref ref="STDOUT" />
+  <!-- Off these ones as they are annoying, and anyway we manage configuration ourself -->
+  <logger name="com.avaje.ebean.config.PropertyMapLoader" level="OFF" />
```
Comment (benmccann) [inline templates/play-scala/conf/logback.xml]: should we reference ebean packages in the scala template since ebean is only in the java template?

File: templates/play-java-intro/conf/logback.xml [hunk 4]
```diff
@@ -1,22 +1,41 @@
+<!-- https://www.playframework.com/documentation/latest/SettingsLogger -->
 <configuration>
 
   <conversionRule conversionWord="coloredLevel" converterClass="play.api.libs.logback.ColoredLevel" />
 
+  <appender name="FILE" class="ch.qos.logback.core.FileAppender">
+     <file>${application.home}/logs/application.log</file>
```
Comment (benmccann) [inline templates/play-java-intro/conf/logback.xml]: I actually rather like not having the FileAppender. Every once in a while I was ending up with a directory being created named "application.home" somehow with this code. Do you know what it is that replaces that variable? Is it built-in to logback or something Play does. I'm not quite sure what the problem I was having was. I think it was either running code in Eclipse and/or running some classes in my project that have their own `main`. Either way the FileAppender never seemed fully fleshed out to me

**Comment by wsargent** [comment]
Okay, how about `${application.home:-.}` instead

File: templates/play-java-intro/conf/logback.xml (hunk 4 above)
Comment (wsargent) [inline templates/play-java-intro/conf/logback.xml]: Okay, how about ${application.home:-.} instead?

Comment (wsargent) [inline templates/play-java-intro/conf/logback.xml]: It should be set in https://github.com/playframework/playframework/blob/2.4.x/framework/src/play/src/main/scala/play/api/Logger.scala#L199

Comment (benmccann) [inline templates/play-java-intro/conf/logback.xml]: I've always found it more annoying than useful, but I'm curious what others feel. Maybe a good compromised would be to add it commented out, so that folks who want it can easily enable it?

Comment (wsargent) [inline templates/play-java-intro/conf/logback.xml]: I'm against anything that requires manual intervention to reflect the default, since this is just doing what Play does with no explicit logback.xml file at all.

File: templates/play-java-intro/conf/logback.xml (hunk 2 above)
Comment (wsargent) [inline templates/play-java-intro/conf/logback.xml]: Filed https://github.com/ebean-orm/avaje-ebeanorm/issues/501

File: templates/play-scala/conf/logback.xml (hunk 3 above)
Comment (wsargent) [inline templates/play-scala/conf/logback.xml]: We only have the one logback default...

File: templates/play-java-intro/conf/logback.xml (hunk 1 above)
Comment (wsargent) [inline templates/play-java-intro/conf/logback.xml]: Well, there's a logback-play-dev.xml, so it might make sense to have logback-play-test.xml as well.  Are you proposing another PR?

https://github.com/playframework/playframework/blob/master/framework/src/play-logback/src/main/resources/logback-play-dev.xml

File: templates/play-java-intro/conf/logback.xml (hunk 2 above)
Comment (wsargent) [inline templates/play-java-intro/conf/logback.xml]: It looks like ebean has resolved this on their end, so we can remove this line.

**Comment by richdougherty** [comment]
LGTM. Since @benmccann has been the main reviewer I'll let him decide whether to push the green button. :)

File: templates/play-java-intro/conf/logback.xml (hunk 2 above)
Comment (benmccann) [inline templates/play-java-intro/conf/logback.xml]: Ok, cool. I just sent a PR to upgrade play-ebean to fix that. See https://github.com/playframework/play-ebean/pull/68

Comment (benmccann) [inline templates/play-java-intro/conf/logback.xml]: I filed an issue to track this: https://github.com/playframework/playframework/issues/5410

File: templates/play-java-intro/conf/logback.xml (hunk 1 above)
Comment (benmccann) [inline templates/play-java-intro/conf/logback.xml]: I was referring to the fact that [logback will automatically look for a file named logback-test.xml](http://logback.qos.ch/manual/configuration.html). I very often want to configure log levels differently in my tests than in I do in dev or production mode. It'd be interesting to show users how to do this. But yeah, maybe that'd be better to do in another PR.

**Comment by benmccann** [comment]
@wsargent I didn't realize that Play had bundled a default logback file inside it, so I didn't quite get what you were doing here at first.

It seems weird that we're having to specify the default in two places now. I like that this change makes the default much more explicit to users. I didn't even know the other one existed. What do you think of deleting the bundled files and then we merge this PR and just use the user-provided one created in the templates?

**Comment by cchantep** [comment]
CI ok for it now

**Comment by wsargent** [comment]
> deleting the bundled files 

That would be okay for the templates, but what about the projects updating from 2.4 to 2.5 that don't have an explicit logback.xml file?

**Comment by benmccann** [comment]
I would think that most people do have a logback.xml file. We have a logback.xml in the 2.4 templates. And even in 2.3 and older we had an application-logger.xml that folks should have renamed to logback.xml in the upgrade to 2.4. I would guess that very few people have no logback.xml.

That being said, we could document it in the migration guide and print out to the console if we don't see a logback.xml or logback.groovy on the classpath.

**Comment by wsargent** [comment]
I'm okay with removing any bundled files completely -- it would certainly simplify the logging logic.

File: templates/play-java-intro/conf/logback.xml [hunk 5]
```diff
@@ -1,22 +1,41 @@
...
 
-  <!--
-    The logger name is typically the Java/Scala package name.
-    This configures the log level to log at for a package and its children packages.
-  -->
+  <appender name="ASYNCFILE" class="ch.qos.logback.classic.AsyncAppender">
+    <appender-ref ref="FILE" />
+  </appender>
+
+  <appender name="ASYNCSTDOUT" gclass="ch.qos.logback.classic.AsyncAppender">
```
Comment (richdougherty) [inline templates/play-java-intro/conf/logback.xml]: `s/gclass/class/`

This typo stops these file from working. If you run the templates you'll see a message like:

```
13:14:46,922 |-ERROR in ch.qos.logback.core.joran.action.AppenderRefAction - Could not find an appender named [ASYNCSTDOUT]. Did you define it below instead of above in the configuration file?
```

You may want to try all the templates and check that logging is working.

**Comment by wsargent** [comment]
@richdougherty fixed the typo, thanks

**Comment by richdougherty** [comment]
LGTM, is this ready to be merged once CI passes?

**Comment by wsargent** [comment]
No idea why this failed...

https://travis-ci.org/playframework/playframework/jobs/106328620#L891

```
[info] 
[info] [info] play.api.Play - Application started (Prod)
[info] java.net.ConnectException: Connection refused
[info] 	at java.net.PlainSocketImpl.socketConnect(Native Method)
[info] 	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
[info] 	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
[info] 	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
[info] 	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
[info] 	at java.net.Socket.connect(Socket.java:589)
[info] 	at java.net.Socket.connect(Socket.java:538)
[info] 	at sun.net.NetworkClient.doConnect(NetworkClient.java:180)
[info] 	at sun.net.www.http.HttpClient.openServer(HttpClient.java:432)
[info] 	at sun.net.www.http.HttpClient.openServer(HttpClient.java:527)
[info] 	at sun.net.www.http.HttpClient.<init>(HttpClient.java:211)
[info] 	at sun.net.www.http.HttpClient.New(HttpClient.java:308)
[info] 	at sun.net.www.http.HttpClient.New(HttpClient.java:326)
[info] 	at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:1169)
[info] 	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1105)
[info] 	at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:989)
[info] 	at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:987)
[info] 	at java.security.AccessController.doPrivileged(Native Method)
[info] 	at java.security.AccessController.doPrivilegedWithCombiner(AccessController.java:782)
[info] 	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:986)
[info] 	at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:933)
[info] 	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1513)
[info] 	at sun.net.www.protocol.http.HttpURLConnection.access$200(HttpURLConnection.java:90)
[info] 	at sun.net.www.protocol.http.HttpURLConnection$9.run(HttpURLConnection.java:1433)
[info] 	at sun.net.www.protocol.http.HttpURLConnection$9.run(HttpURLConnection.java:1431)
[info] 	at java.security.AccessController.doPrivileged(Native Method)
[info] 	at java.security.AccessController.doPrivilegedWithCombiner(AccessController.java:782)
[info] 	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1430)
[info] 	at java.net.URL.openStream(URL.java:1045)
[info] 	at sbt.Using$$anonfun$urlReader$1.apply(Using.scala:80)
[info] 	at sbt.Using$$anonfun$urlReader$1.apply(Using.scala:80)
[info] 	at sbt.Using$$anon$3.open(Using.scala:59)
[info] 	at sbt.Using.apply(Using.scala:23)
[info] 	at sbt.IO$.readLinesURL(IO.scala:684)
[info] 	at $e11f993142a47ff8ec94$$anonfun$$sbtdef$1$$anonfun$apply$2$$anonfun$apply$1$$anonfun$1.apply(/tmp/sbt_3d814b3f/distribution/build.sbt:55)
[info] 	at $e11f993142a47ff8ec94$$anonfun$$sbtdef$1$$anonfun$apply$2$$anonfun$apply$1$$anonfun$1.apply(/tmp/sbt_3d814b3f/distribution/build.sbt:55)
[info] 	at $129a1634dabdf988b450$.retry(build.sbt:39)
[info] 	at $129a1634dabdf988b450$.retry(build.sbt:46)
[info] 	at $129a1634dabdf988b450$.retry(build.sbt:46)
[info] 	at $129a1634dabdf988b450$.retry(build.sbt:46)
[info] 	at $129a1634dabdf988b450$.retry(build.sbt:46)
[info] 	at $129a1634dabdf988b450$.retry(build.sbt:46)
[info] 	at $129a1634dabdf988b450$.retry(build.sbt:46)
[info] 	at $129a1634dabdf988b450$.retry(build.sbt:46)
[info] 	at $129a1634dabdf988b450$.retry(build.sbt:46)
[info] 	at $129a1634dabdf988b450$.retry(build.sbt:46)
[info] 	at $e11f993142a47ff8ec94$$anonfun$$sbtdef$1$$anonfun$apply$2$$anonfun$apply$1.apply$mcV$sp(/tmp/sbt_3d814b3f/distribution/build.sbt:54)
[info] 	at $e11f993142a47ff8ec94$$anonfun$$sbtdef$1$$anonfun$apply$2$$anonfun$apply$1.apply(/tmp/sbt_3d814b3f/distribution/build.sbt:51)
[info] 	at $e11f993142a47ff8ec94$$anonfun$$sbtdef$1$$anonfun$apply$2$$anonfun$apply$1.apply(/tmp/sbt_3d814b3f/distribution/build.sbt:51)
[info] 	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
[info] 	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
[info] 	at sbt.std.Transform$$anon$4.work(System.scala:63)
[info] 	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
[info] 	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
[info] 	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
[info] 	at sbt.Execute.work(Execute.scala:235)
[info] 	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
[info] 	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
[info] 	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
[info] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
[info] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[info] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[info] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[info] 	at java.lang.Thread.run(Thread.java:745)
[info] [error] (*:checkConfig) java.net.ConnectException: Connection refused
[info] [error] Total time: 5 s, completed Feb 1, 2016 9:56:08 PM
[error] x play-sbt-plugin / distribution
[error]    {line 34}  Command failed: checkConfig failed
```

**Comment by wsargent** [comment]
> LGTM, is this ready to be merged once CI passes?

The only thing I would change is from `${application.home}` to `${application.home:-.}` but I can do that in another PR.

**Comment by mkurz** [comment]
@wsargent Where is the syntax of `${application.home:-.}` documented?

**Comment by wsargent** [comment]
@mkurz Created https://github.com/playframework/playframework/issues/5535 and the reference is in there.
